# Analyzing CIA Factbook Data Using SQL
The CIA World Factbook contains demographic data of all the world's countries. Today, I will use a the CIA factbook dataset available <a href='https://dsserver-prod-resources-1.s3.amazonaws.com/257/factbook.db'>here</a>.

I will use SQL to analyze the dataset and perform some mathematical and statistical analysis of the data.

In [1]:
%%capture
%load_ext sql

In [2]:
%sql sqlite:////Users/admin/Downloads/Databases/factbook.db    

'Connected: @/Users/admin/Downloads/Databases/factbook.db'

In [3]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table';

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [4]:
%%sql
SELECT *
  FROM facts
 LIMIT 5;

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

## Data Analysis (a)
The "facts" table contains the data I am interested in. I will perform some mathematical and statistical analysis of the population and population growth columns.

In [5]:
%%sql
SELECT MIN(population), MAX(population), MIN(population_growth), MAX(population_growth)
  FROM facts;

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

In [6]:
%%sql
SELECT name
  FROM facts
 WHERE population = (SELECT MIN(population)
                       FROM facts);

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[('Antarctica',)]

In [7]:
%%sql
SELECT name
  FROM facts
 WHERE population = (SELECT MAX(population)
                      FROM facts);

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[('World',)]

## Data Analysis (b)
The World is included in the dataset. Since this is not a country but actually the sum total of all the population, land, and water on earth, I will remove this from the dataset for my further analysis.

In [8]:
%%sql
SELECT MIN(population), MAX(population), MIN(population_growth), MAX(population_growth)
  FROM facts
 WHERE name != 'World';

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[(0, 1367485388, 0.0, 4.02)]

In [9]:
%%sql
SELECT AVG(population), AVG(area)
  FROM facts
 WHERE name != 'World';

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[(32242666.56846473, 555093.546184739)]

In [10]:
%%sql
SELECT name
  FROM facts
 WHERE population > (SELECT AVG(population)
                       FROM facts)
   AND area < (SELECT AVG(area)
                 FROM facts);

 * sqlite:////Users/admin/Downloads/Databases/factbook.db
Done.


[('Bangladesh',),
 ('Germany',),
 ('Japan',),
 ('Philippines',),
 ('Thailand',),
 ('United Kingdom',),
 ('Vietnam',)]

## Conclusion
The average population per country is about 32,000,000 people. The highest population of any country is 1,367,485,388 people. 7 countries are very dense, having populations higher than the world average while simultaneously having land areas less than the world average.

Further exploration could include determining which countries have dense population growth, which I will define by having higher than average population growth and a simultaneously lower than average land area.